src/com/senlainc/warsaw/tyurin/service/GaragePlaceService.java

```diff
@@ -0,0 +1,130 @@
+package com.senlainc.warsaw.tyurin.service;
+
+import com.senlainc.warsaw.tyurin.dao.CraftsmanDAO;
+import com.senlainc.warsaw.tyurin.dao.GaragePlaceDAO;
+import com.senlainc.warsaw.tyurin.dao.ICraftsmanDAO;
+import com.senlainc.warsaw.tyurin.dao.IGaragePlaceDAO;
+import com.senlainc.warsaw.tyurin.entity.GaragePlace;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public class GaragePlaceService implements IGaragePlaceService{
+
+    private static GaragePlaceService INSTANCE;
+    private IGaragePlaceDAO garagePlaceDAO;
+    private ICraftsmanDAO craftsmanDAO;
```

Comment:
Unfortunately, it is bad practice to use someone else's dao in an inappropriate service. If you really need to work with the mechanics in this service, you can do it through the mechanics service, not through the dao.

Reply:
But think about pass desired mechanic to method what need it, if it's possible.